<a href="https://colab.research.google.com/github/ahmedmoustafa/AUC-DSCI1412-Spring2024/blob/main/examples/maternal-smoking/smoking.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Maternal Smoking

First, we need to import the packages (libraries) that will be used in the analysis, [Pandas](https://pandas.pydata.org/) for data loading and wrangling, and [Seaborn](https://seaborn.pydata.org/) for data visualization.

In [1]:
import pandas as pd # Import the pandas package using the alias 'pd'
import seaborn as sns # Import the seaborn package using the alias 'sns'
import numpy as np # Import the numpy package using the alias 'np'
import scipy.stats as stats # Import the scipy.stats package using the alias 'stats'

Now, we can easily access the data from an online [`TSV`](https://en.wikipedia.org/wiki/Tab-separated_values) file using Pandas' [read_table()](https://pandas.pydata.org/docs/reference/api/pandas.read_table.html) function. Before moving forward, you may want to open and examine the data file at this link: [https://tinyurl.com/mpd4jx9n](https://tinyurl.com/mpd4jx9n). This will help you gain a better understanding of the data's structure and content.

In [2]:
df = pd.read_table("https://raw.githubusercontent.com/ahmedmoustafa/datasets/main/smoking/smoking.tsv") # Read and load the tsv data file into a DataFrame
df.shape

(610, 20)

The data got loaded successfully into a Pandas [`DataFrame`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html) variable called `df`. The [`shape`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.shape.html) of the `dataframe` shows that there are `610` rows (records) and `20` columns.

Let's take a quick look at the data using the [`head()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.head.html) function.

In [3]:
df.head()

id  date  gestation  weight  parity mom.race  mom.age  mom.edu  \
0   15  1411        284     120       1    asian       27        5   
1   20  1499        282     113       2    white       33        5   
2  100  1673        286     136       4    white       25        2   
3  129  1562        245     132       2    black       23        1   
4  142  1408        289     120       3    white       25        4   

   mom.height  mom.weight dad.race  dad.age  dad.edu  dad.height  dad.weight  \
0          62         100    asian       31        5          65         110   
1          64         135    white       38        5          70         148   
2          62          93    white       28        2          64         130   
3          65         140    black       23        4          71         192   
4          62         125    white       26        1          70         180   

   marital  income            smoke  quit.time  cigs  
0        1       1            never          0     0  
1        1       4            never          0     0  
2        1       4  until_pregnancy          2     2  
3        1       2            never          0     0  
4        0       2            never          0     0

---